nemo/collections/nlp/models/language_modeling/transformer_lm_model.py

```diff
@@ -142,145 +119,154 @@ def training_step(self, batch, batch_idx):
         passed in as `batch`.
         """
         # forward pass
-        input_ids, input_mask, labels = batch
+        for i in range(len(batch)):
+            if batch[i].ndim == 3:
+                # Dataset returns already batched data and the first dimension of size 1
+                # added by DataLoader is excess.
+                batch[i] = batch[i].squeeze(dim=0)
+        ids, mask = batch
+        input_ids, labels = ids[:, :-1], ids[:, 1:]
+        input_mask = mask[:, :-1]
         log_probs = self(input_ids=input_ids, attention_mask=input_mask)
 
-        train_loss = self.training_loss(log_probs=log_probs, labels=labels)
-        training_perplexity = self.training_perplexity(logits=log_probs)
+        train_loss = self.loss_fn(log_probs=log_probs, labels=labels)
 
         tensorboard_logs = {
             "train_loss": train_loss,
             "lr": self._optimizer.param_groups[0]["lr"],
-            "train_ppl": training_perplexity,
         }
         return {"loss": train_loss, "log": tensorboard_logs}
 
+    def eval_step(self, batch, batch_idx, mode):
+        for i in range(len(batch)):
+            if batch[i].ndim == 3:
+                # Dataset returns already batched data and the first dimension of size 1
+                # added by DataLoader is excess.
+                batch[i] = batch[i].squeeze(dim=0)
+        ids, mask = batch
+        input_ids, labels = ids[:, :-1], ids[:, 1:]
+        input_mask, output_mask = mask[:, :-1], mask[:, 1:]
+        log_probs = self(input_ids=input_ids, attention_mask=input_mask)
+
+        eval_loss = self.loss_fn(log_probs=log_probs, labels=labels)
```

Comment:
We should probably have a non-label smoothed cross-entropy for the eval_step. I just made this change for NMT where `self.loss_fn` is smoothedCE and `self.eval_loss_fn` is NLLLoss.